Mathlib/Tactic/Simproc/Divisors.lean

```diff
@@ -0,0 +1,48 @@
+import Mathlib.NumberTheory.Divisors
+import Mathlib.Lean.ToExpr
+import Mathlib.Data.Finset.Sort
+import Lean
+
+open Lean Meta
+
+simproc_decl Nat.divisorsEq (Nat.divisors _) := fun e => do
+  unless e.isAppOfArity `Nat.divisors 1 do return .continue
+  let some n ← fromExpr? e.appArg! | return .continue
+  --TODO: do we want to define `Nat.divisorsList` and use this instead of `n.divisors.sort (· ≤ ·)`?
+  let rhs ← (n.divisors.sort (· ≤ ·)).toFinsetExpr (α := ℕ)
+  /- The last two lines can be replace by `return .done {expr := rhs }` since the proof is `rfl`.
+  Not sure what's best here. -/
+  let pf ← Meta.mkDecideProof (← mkEq e rhs)
```

Comment:
ooh, I think there might be a third option (the first being rfl, the second being decide) which could be faster here, will look into this once it's daytime

Reply:
resolving this for now; there are faster ways to make proof terms for this but those can wait for now

Reply:
Actually, perhaps it's worth putting a TODO in this file to investigate performance improvements to this. One much faster option would be to compute the list of divisors in the VM, produce a proof term that the number of divisors is equal to the length of that list (using multiplicativity), produce a proof term that everything in the list is a divisor, and combine these to show that the list of divisors is exactly what it should be. That should be much faster in practice (still linear, but the O(n) part is happening in the VM and the rest is O(n^eps) for small eps); but it's quite a bit more work to do, so leaving a TODO for this seems reasonable.

Reply:
That's a cool idea! I've added a note pointing to the discussion here.